ui-components/src/text/Heading.tsx

```diff
@@ -10,13 +10,13 @@ export interface HeadingProps {
 export type HeaderType = keyof typeof HeaderStyleMap
 
 const HeaderStyleMap = {
-  cardHeader: "card-header",
-  cardSubheader: "card-subheader",
-  tableHeader: "table-header",
-  tableSubheader: "table-subheader",
-  sidebarHeader: "text-caps-underline",
-  categoryHeader: "category-header",
-  sidebarSubHeader: "text-caps-tiny",
+  largeHeader: "text__large-header",
+  mediumHeader: "text__medium-header",
+  smallWeighted: "text__small-weighted",
+  smallNormal: "text__small-normal",
+  underlineHeader: "text__underline-header",
+  lightWeighted: "text__light-weighted",
+  capsWeighted: "text__caps-weighted",
 }
 
 const Heading = (props: HeadingProps) => {
```

Comment:
I know it would be another breaking change, but a while back we'd discussed changing the `style` prop to `styleType` to match some other components in the system like Button, and also so it's not confused with the HTML `style` prop/attribute. If we're making sweeping changes to `Header` anyway, this feels like the opportune moment!

Reply:
@emilyjablonski Thoughts on the above?